Update Suredbits Wallet

Christewart opened the pull request:
Update Suredbits Wallet to these docker images

Frontend: 

https://hub.docker.com/layers/wallet-server-ui/bitcoinscala/wallet-server-ui/1.9.2-f44ad4a4-SNAPSHOT/images/sha256-f61ce1c3acc28c519c3595d0984679d5becbacd1d4aaba2599ad39999f2e9ba8?context=explore

Backend: 

https://hub.docker.com/layers/bitcoin-s-server/bitcoinscala/bitcoin-s-server/1.9.2-66-b16a8ca6-SNAPSHOT/images/sha256-b46938aba25cf08af409d68b548347d71888454e28185d589a1f42911b1ac2f5

nevets963 reviewed (approved):
@Christewart Thanks for the app update! I've bumped the version in the app manifest to include the hash of the `wallet-server-ui` Docker image (rightly or wrongly). If that version number doesn't change, then existing users are not alerted to the new app update. Or is this simply the exact same Docker image but you only changed the tag to a new convention?

I've tested on AMD64 and ARM64 and I'm OK to approve this PR.

@lukechilds Do you want to test too?

Christewart commented:
> @Christewart Thanks for the app update! I've bumped the version in the app manifest to include the hash of the `wallet-server-ui` Docker image (rightly or wrongly). If that version number doesn't change, then existing users are not alerted to the new app update. Or is this simply the exact same Docker image but you only changed the tag to a new convention?
> 
> I've tested on AMD64 and ARM64 and I'm OK to approve this PR.
> 
> @lukechilds Do you want to test too?

Hey @nevets963 I found a bug late last night in this release that I'm going to get fixed today on the backend server. Please hold off on merging on until its fixed :pray: 

>EDIT: Please do not merge until we have this bug fix: https://github.com/bitcoin-s/bitcoin-s/issues/4455

Christewart commented:
@nevets963 This should be good again

Christewart commented:
> @Christewart Ok cool, I'll test again, shall we go with `1.9.2-f44ad4a4` as the version within the app manifest file?

We will have to start doing DLCs on "Does Suredbits ship a working docker image to the umbrel team?" ;)

nevets963 commented:
Haha, let's do it! :P I'll also get this tested in the morning :)

Christewart commented:
> Haha, let's do it! :P I'll also get this tested in the morning :)

Can you also look at the authentication on this PR? Its unclear to me how authentication works. I was told "it just works", but this morning i was able to open my Suredbits Wallet app without having authenticated on the `umbrel.local` page this morning.

Christewart commented:
Maybe our DLC can be based on "Did Suredbits do umbrel authentication right?" If so, i'll send you an offer if you give me your tor address. I'll bet that we did do it right unless you want to take that side. Maybe a small wager of 10k sats?

You can obtain your tor address by clicking `main` and then copying the tor address to your clipboard and sending it to me.

![Screenshot from 2022-07-08 08-09-46](https://user-images.githubusercontent.com/3514957/177999390-5d584beb-e0e5-49b5-a9d1-6550246078b1.png)
![Screenshot from 2022-06-30 10-21-30](https://user-images.githubusercontent.com/3514957/177999392-a5f950f8-2feb-460c-8917-aa253063f76c.png)

Christewart commented:
hi @nevets963 any updates on this?

lukechilds commented:
@nevets963 is currently on leave, I'll be aiming to get all the outstanding app PRs merged this week.

lukechilds commented:
Just double checking, @Christewart you have this pinned to the correct snapshots, right?

lukechilds commented:
Also we have a new `releaseNotes` property in `umbrel-app.yml` that you can use to show a short description to the user of what the update includes.

lukechilds commented:
@rorp @user411 @Christewart just a friendly ping on this incase you missed it.

Christewart commented:
> Also we have a new `releaseNotes` property in `umbrel-app.yml` that you can use to show a short description to the user of the changes included in this update.

I added the `releaseNotes` property to `umbrel-app.yml`. Lmk if I did it wrong.

> Just double checking, @Christewart you have this pinned to the correct snapshots, right?

I corrected the snapshots I had in the OP to be consistent with the contents of the PR.

lukechilds commented on suredbits-wallet/docker-compose.yml:
```diff
@@ -7,7 +7,7 @@ services:
 
   web:
-    image: bitcoinscala/wallet-server-ui:1.9.2@sha256:8d53cb8bc595d1f55423b63fe40a57728cf359200a3f4409e809b20526833c28
+    image: bitcoinscala/wallet-server-ui:1.9.2-f44ad4a4-SNAPSHOT
```
```suggestion
    image: bitcoinscala/wallet-server-ui:1.9.2-f44ad4a4-SNAPSHOT@sha256:61d0f156202314cdfe00e184affa72f98c23fa79b8c9f56eed93f5802b0c2f82
```

lukechilds commented on suredbits-wallet/docker-compose.yml:
```diff
@@ -30,7 +30,7 @@ services:
   
   walletserver:
-    image: bitcoinscala/bitcoin-s-server:1.9.2@sha256:b042c7c32fab2157f93612915f8579695cf1e4d471cdcc32d7eb7aecb0234c34
+    image: bitcoinscala/bitcoin-s-server:1.9.2-66-b16a8ca6-SNAPSHOT
```
```suggestion
    image: bitcoinscala/bitcoin-s-server:1.9.2-66-b16a8ca6-SNAPSHOT@sha256:bde3cbb50c4e5e8e4a5f28ae90e84718f093506e4f8ae6e9c3d0894bd7e102f5
```

lukechilds commented:
Ok this is live now! 🚀

Would be great if you could confirm over here too for Krystal Bull: https://github.com/getumbrel/umbrel-apps/pull/64#issuecomment-1197958429

lukechilds commented:
>Can you also look at the authentication on this PR? Its unclear to me how authentication works. I was told "it just works", but this morning i was able to open my Suredbits Wallet app without having authenticated on the umbrel.local page this morning.

Yeah don't worry about this. We currently have two auth systems running in parallel in Umbrel which is a bit awkward. We have the old legacy JWT based auth for the Umbrel dashboard and the new session cookie based auth for the proxy in front of the apps. We've linked them together in some places so e.g logging out of the dashboard nukes the app proxy session cookies. But currently it's possible for the dashboard JWT to timeout but the app proxy session cookies won't be timed out, so it appears that you can bypass auth, but really you were already authed and there was just a clientside expiry of the JWT token for the dashbaord and not the session cookie for the app proxy.

It's a temporary issue as we're incrementally refactoring the auth, we plan to pull all the old JWT auth logic out of the dashboard and just have the Umbrel dashboard sitting behind the app proxy like all other apps. Then it'll be a single auth system for everything and Umbrel dashboard and app auth state will always be in sync.
